disable_deactivated_accounts.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+import sys
+import optparse
+import requests
+import ldap
+import json
+
+from auth0_api_settings import AUTH0_API_KEY, AUTH0_API_URL, AUTH0_CONNECTION, LDAP_HOST,\
+    LDAP_USER, LDAP_PASS, EXCLUSION_LIST, USE_PROXY, PROXIES
+
+# function to build headers that'll be used by all API calls
+def build_headers():
+    headers = {}
+    headers['Authorization'] = 'Bearer %s' % AUTH0_API_KEY
+    headers['Content-Type'] = "application/json"
+    headers['Accept'] = "application/json"
+    return headers
+
+# generic function to just get all users belonging to the specific enterprise connection
+# it has some logic to page through a list longer than 50, stolen from the Okta version of this script, needs tested once we have more users
+def list_all_users():
+    should_return = False
+    return_list = []
+    fetch_url = "%s/users" % (AUTH0_API_URL)
+    payload = {'connection': "%s" % AUTH0_CONNECTION, 'fields': 'user_id,email,blocked', 'include_fields': 'true'}
+    while (should_return is False):
+        prev_url = fetch_url
+        if USE_PROXY is True:
+            resp = requests.get(
+                fetch_url,
+                headers=build_headers(),
+                params=payload,
+                proxies=PROXIES
+            )
+        else:
+            resp = requests.get(fetch_url, headers=build_headers(), params=payload)
+            req = resp.request
+        return_list += resp.json()
+        try:
+            fetch_url = resp.links['next']['url']
+            if fetch_url == '' or fetch_url is None or fetch_url == prev_url:
+                should_return = True
+        except (KeyError, IndexError):
+            should_return = True
+    return return_list
+
+
+# filter out only the active users
+# active user here is defined as not being currently blocked
+def list_all_active_users():
+    users = list_all_users()
+    return [user for user in users if user[u'blocked'] != True]
```

Comment:
I added another condition to catch users that don't have the blocked attribute at all (users that have never been blocked, won't have it). In the latest commit, I re-wrote this in long form to be more readable.